processing/src/main/java/org/apache/druid/query/InlineDataSource.java

```diff
@@ -300,7 +363,7 @@ public boolean equals(Object o)
       return false;
     }
     InlineDataSource that = (InlineDataSource) o;
-    return rowsEqual(rows, that.rows) &&
```

Comment:
maybe compare frames directly if possible and save on the cost of initializing various frameReaders and then de serializing them to rows.

Reply:
I'll check if this works since that would be faster.
InlineDatasource's equality is done only in the tests afaik, therefore we should be fine if it doesn't work as such.